src/sidebar/store/modules/test/groups-test.js

```diff
@@ -108,4 +139,140 @@ describe('sidebar.store.modules.groups', () => {
+          'If logged out and there is an open group show it in `Featured Groups`.',
+        isLoggedIn: false,
+        filteredGroups: [openGroup],
+        expectedGroups: [openGroup],
+      },
+    ].forEach(({ description, isLoggedIn, filteredGroups, expectedGroups }) => {
+      it(description, function() {
+        store.getState().session = { userid: isLoggedIn ? '1234' : null };
+        store.loadGroups(filteredGroups);
+
+        const featuredGroups = store.getFeaturedGroups();
+        const currentlyViewing = store.getCurrentlyViewingGroups();
+
+        assert.isTrue(featuredGroups.length === expectedGroups.length);
+        let i;
+        for (i = 0; i < featuredGroups.length; ++i) {
+          assert.isTrue(featuredGroups[i].id === expectedGroups[i].id);
+        }
+        assert.isTrue(currentlyViewing.length === 0);
+      });
+    });
+  });
+
+  describe('getMyGroups', () => {
+    [
+      {
+        description:
+          'If logged in and the user is a member of the group show it in `My Groups`',
+        isLoggedIn: true,
+        filteredGroups: [openGroup],
+        expectedGroups: [],
+      },
+      {
+        description:
+          'If logged in and the user is a member of the unscoped group show it in `My Groups`',
+        isLoggedIn: true,
+        filteredGroups: [restrictedUnscopedMemberGroup],
+        expectedGroups: [restrictedUnscopedMemberGroup],
+      },
+      {
+        description:
+          'If logged in and the user is a member of the group show it in `My Groups`',
+        isLoggedIn: true,
+        filteredGroups: [publicGroup],
+        expectedGroups: [publicGroup],
+      },
+    ].forEach(({ description, isLoggedIn, filteredGroups, expectedGroups }) => {
+      it(description, function() {
+        store.getState().session = { userid: isLoggedIn ? '1234' : null };
+        store.loadGroups(filteredGroups);
+
+        const myGroups = store.getMyGroups();
+        const currentlyViewing = store.getCurrentlyViewingGroups();
+
+        assert.isTrue(myGroups.length === expectedGroups.length);
+        let i;
+        for (i = 0; i < myGroups.length; ++i) {
+          assert.isTrue(myGroups[i].id === expectedGroups[i].id);
+        }
+        assert.isTrue(currentlyViewing.length === 0);
+      });
+    });
+  });
+
+  describe('getCurrentlyViewingGroups', () => {
+    [
+      {
+        description:
+          'If logged out and there is a restricted group not scoped to the page, show it in `Currently Viewing`',
+        isLoggedIn: false,
+        filteredGroups: [restrictedUnscopedGroup],
+      },
+      {
+        description:
+          'If logged out and only the Public group is present, show it in `Currently Viewing`',
+        isLoggedIn: false,
+        filteredGroups: [publicGroup],
+      },
+      {
+        description:
+          'If logged in and there is a restricted group not scoped to the page of which the user is not a member, show it in `Currently Viewing`',
```

Comment:
Ooof...can't make heads or tails of this description! :D

Is it "If logged in and there is an out-of-scope restricted group to which the user is not a member, it should be included in the Currently-Viewing list"?